llvm/lib/DebugInfo/DWARF/DWARFUnit.cpp

```diff
@@ -751,7 +751,7 @@ void DWARFUnit::updateAddressDieMap(DWARFDie Die) {
 DWARFDie DWARFUnit::getSubroutineForAddress(uint64_t Address) {
   extractDIEsIfNeeded(false);
   if (AddrDieMap.empty())
-    updateAddressDieMap(getUnitDIE());
+    updateAddressDieMap(getNonSkeletonUnitDIE(/*ExtractUnitDIEOnly=*/false));
```

Comment:
this seems interesting - what motivates this change/could it be separated from the basic "dump the non-dwos" stuff? (`getNonSkeletonUnitDIE` unconditionally could cause a lot more DWARF parsing/trying to read all the DIES in non-skeleton units too when they weren't being read before this patch?)

Reply:
This was motivated by the `--lookup <addr>' stuff. If we have a skeleton compile unit, wouldn't we always want to try and grab the .dwo file and do a real lookup instead of just failing? Or worse yet, we have the .dwo file and we are doing a lookup, but the compiler option that leaves inlines in the skeleton compile unit has been enabled and you get subpar results.

Reply:
I'd err on the side of not searching all the dwos without opting in, since it might be quite expensive/a lot of results or things to search through. In any case, seems best to discuss in a separate review, perhaps.

I worry about whether this codepath affects symbolizing in some way, for instance (which the inline info in the skeleton is meant to help with/avoid having to parse the dwos). Separate patches will let us track regressions more carefully.